clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceAfterImageConsumerImpl.java

```diff
@@ -173,13 +173,17 @@ protected CompletableFuture<Void> internalSeekToEndOfPush(Set<Integer> partition
           LOGGER.info(
               "Seeking to EOP for partitions: " + partitions.toString() + " for version topic: "
                   + targetTopic.getName());
+          subscriptionLock.writeLock().lock();
+          lockAcquired = true;
```

Comment:
Should we use tryLock with infinite timeout and use that var when releasing lock?